### ISSUE TITLE
Support power timers should not reset on capture

### ISSUE THREAD

**Issue by kyrylo** [OpenRA/OpenRA#10732]
How it works right now:

1. Capture someone's missile silo
2. Evidence the nuke timer reset

How It should be:

1. Capture someone's missile silo
2. Evidence that the nuke timer is transferred to you, so you can continue building it

The same should apply to Chronosphere, GPS and Iron Curtain.

Thoughts?

**Comment by pchote** [comment]
Support powers belong to the player, just like production queues.  The idea falls apart as soon as you need to consider structures that don't have a build limit: what happens with both player's GPS timers if someone steals one of your two tech centers? or if you steal someone else's while you also have your own?

**Comment by kyrylo** [comment]
> what happens with both player's GPS timers if someone steals one of your two tech centers? or if you steal someone else's while you also have your own?

How about:

* if you stole it and you don't have a tech center, then you start producing GPS from where it was left
* if you stole it and you already have a tech center/GPS, then you get a speed bonus to your own timer (say, shave off 4 minutes on Normal speed) and the player who lost one of their tech centers keeps producing their own GPS like nothing happened.

**Comment by GraionDilach** [comment]
No, no, definitely not! This is a really bad idea. :-1:

**Comment by Micr0Bit** [comment]
i've posted this idea yesterday ... and it would make sense ( i think @penev92 agreed to that)
well what would happen if someone is stealing a techcenter ... while his own gps is up ... they both get treated separately... the new gps-counter shows up (even though there is allready  a gps up) 

the only benefit (in this particular case) would be the 2 gps-satellites ... if one techcenter is down , you still have another one ... therefor , you dont loose gps ... 

but 

we currently dont loose the gps-satellites when the techcenter is destroyed ... as soon as we build another tech-center , the gps instantly gets enabled again ... (i consider that a little bit OP ... cause the techcenter isnt even limited to build , you can spam them )

this is no bad idea at all , makes totally sense to me ... and would help the balance (gps related)

and this way , we could allow players to own multiple "nukes" while still having the "build-nuke-silo" limitation

**Comment by Micr0Bit** [comment]
oh an , im not super-interested in having this in ... but this restriction seems a bit unfair to me , considering that the players allready have to deal with "building-limitation" ... and i know there is some people that would be really happy about having this 

not that it would make a lot of difference ... cause capturing enemys tech-buildings gets quickly noticed anyway ... but still

**Comment by vietman391** [comment]
@GraionDilach I don't think we should be dismissing the idea entirely, and despite the fact that this may seem overpowered, it's very likely to be a rare case in-game either way. Since when do engineers ever get a cap-off of enemy buildings, especially heavily defended ones like missile silos in the first place? Even then, high-level players would sell their high-tech buildings before they get captured, and it doesn't really matter at all if it's a 1v1.

@pchote, I don't understand why you say "the idea falls apart," as you could've just posed the exception of the multiple tech centers, which @Micr0Bit just addressed/solved. Are all programmers that picky about details?

Anyways, I think @kyrylo makes a good point here. Allowing these types of engineer abilities would give players a larger incentive to use engineers more strategically, rather than just having them as behind-the-scene units to take your oil derricks for you. In a competitive match though, this proves to be less useful due as high-level players are typically more attentive. So...why not? Can't you just let the noobs have fun for once? :P 

- Spy

**Comment by GraionDilach** [comment]
Resetting a timer on a super/support weapon is a __standard RTS feature in all RTSes__ I've played. If this is required to maintain balance than I'll clearly say you're trying to justify this to sweep underlying issues under the rug.

The "timer bonus for owning multiple structures of the same type" idea just adds to this. This idea just confuses players and encourages SW stunlock.

**Comment by vietman391** [comment]
@GraionDilach I don't believe you saw my previous argument, "Since when do engineers ever get a cap-off of enemy buildings, especially heavily defended ones like missile silos in the first place? Even then, high-level players would sell their high-tech buildings before they get captured, and it doesn't really matter at all if it's a 1v1."

What do you mean by "standard RTS feature"? Starcraft, for instance, is a pivotal series in the RTS industry, and has no way of capturing buildings. So how does that make it a standard feature if it doesn't exist in a game like that?

I'm not saying @kyrylo's suggested feature is required for balance, nor did he say that as well. I'm saying it is a reasonable and not at all irrational addition to the fun that we already have in OpenRA.

I'd also like to know what these "underlying issues" you are alluding to.

**Comment by Micr0Bit** [comment]
well lets skip the "add a speed bonus" idea (kyrylo allready said , "probably unrelated)

**Comment by pchote** [comment]
@vietman391: You're trying to justify the downsides of the feature by saying that this feature is not going to affect anything in the vast majority of games.  In that case, why should we spend effort on changing the feature at all?

**Comment by anjew175** [comment]
He is trying to say the downsides won't negatively effect games and is an interesting feature for a modern-day Red Alert. The chance of succeeding with a capture on a tech building when you havent won the game is very very **very **small (higher chances with britain + phase but all you need is 1 rifle to kill the engi while the phase tries to run you over)

**Comment by fgdzfggsfgsfgsf** [comment]
In theory I like this idea.

In practice though, these buildings are usually only captured if a player disconnects, and this would make capturing a disconnected player's stuff even more "OP", and severely punishes players who are far away from the disconnected player.

**Comment by KOYK** [comment]
still this is a good trait for other mods.

**Comment by PunkPun** [comment]
Do we call this a won't-fix for base mods?
